tests/TestCase/Controller/Component/UserPermissionComponentTest.php

```diff
@@ -176,4 +177,59 @@ public function testUserWithPermissionAndTrueCallback() {
 
         $this->assertEquals($expected, $result);
     }
+	
+	public function testMissingHandlerBeingIgnored() {
+		$userType = "user";
+		$action = "action";
+		
+		$rules = array(
+			"user_type" => $userType,
+			"redirect" => "",
+			"message" => "You don't have permission to access this page",
+			"action" => $action,
+			"controller" => $this->controller,
+			"groups" => array(
+				$userType => array($action)
+			),
+			"views" => array(
+				$action = "handlerThatDoesNotExistForSure"
+			)
+		);
+		
+		$result = $this->userPermissions->allow($rules);
+		$expected = false;
+		
+		$this->assertEquals($expected, $result);
+	}
+	
+	public function testMissingHandlerThrowsException() {
+		$this->userPermissions->initialize(array("throwEx" => true));
+		$userType = "user";
+		$action = "action";
+		
+		$rules = array(
+			"user_type" => $userType,
+			"redirect" => "",
+			"message" => "You don't have permission to access this page",
+			"action" => $action,
+			"controller" => $this->controller,
+			"groups" => array(
+				$userType => array($action)
+			),
+			"views" => array(
+				$action = "handlerThatDoesNotExistForSure"
+			)
+		);
+		
+		try {
+			$result = $this->userPermissions->allow($rules);
+			$this->fail("No exception thrown");
+		}
+		catch(MissingHandlerException $e) {
+			
+		}
+		catch(\Exception $e) {
+			$this->fail("Wrong exception thrown " . get_class($e));
```

Comment:
In this case is better to use annotation for example read this:

https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.expectedException